readthedocs/projects/tasks.py

```diff
@@ -271,6 +361,7 @@ def run(self, version_pk):  # pylint: disable=arguments-differ
             with environment:
                 before_vcs.send(sender=self.version, environment=environment)
                 with self.project.repo_nonblockinglock(version=self.version):
+                    self.pull_cached_environment()
```

Comment:
Do we not `push` it back up during `sync`?

Reply:
I did this on purpose. Since we have per-version cached environment and sync operation is not tied to a version --well, it's tied but we always pass the default one.

The other problem could be a race condition. Even if we would pass different versions to sync, it may conflict with the build itself trying to both push the cache to the storage at the same time or even sync after build (loosing the real cache)

> I could add this explanation as a comment in the code, probably